src/app/(proper_react)/redesign/(authenticated)/user/dashboard/subscribed/page.tsx

```diff
@@ -2,76 +2,49 @@
-import { headers } from "next/headers";
-
-export default async function Subscribed() {
-  const session = await getServerSession(authOptions);
-  if (!session?.user?.subscriber) {
-    throw new Error("No session");
-  }
-
-  if (!(await isEligibleForPremium(session.user, getCountryCode(headers())))) {
-    throw new Error("Not eligible for premium");
-  }
-
-  const result = await getOnerepProfileId(session.user.subscriber.id);
-  const profileId = result[0]["onerep_profile_id"] as number;
-
-  try {
-    await activateProfile(profileId);
-    await optoutProfile(profileId);
-  } catch (ex) {
-    console.debug(ex); // TODO handle
-  }
-
-  const dev =
-    process.env.NODE_ENV === "development" || process.env.APP_ENV === "heroku";
-  const latestScan = await getLatestOnerepScanResults(profileId);
-  if (!latestScan.scan) {
-    throw new Error("Must have performed manual scan");
-  }
-
-  const scans = await getAllScanResults(profileId);
-
-  // In dev mode, record scans every time this page is reloaded.
-  // The webhoook does this in production.
-  if (dev) {
-    await addOnerepScanResults(
-      profileId,
-      latestScan.scan.onerep_scan_id,
-      scans,
-      "initial"
-    );
-  }
-
-  return (
-    <div>
-      <div>
-        <h3>You are now subscribed</h3>
-        {dev ? (
-          <div>
-            <h3>Dev mode enabled</h3>
-            <p>Reload this page to update scan</p>
-            <pre>{JSON.stringify(scans, null, 2)}</pre>
-          </div>
-        ) : (
-          ""
-        )}
-      </div>
-    </div>
-  );
+"use client";
+
+import { useSession } from "next-auth/react";
+import { useEffect } from "react";
+import { useRouter } from "next/navigation";
+
+import { hasPremium } from "../../../../../../functions/universal/user";
+import { captureException } from "@sentry/browser";
+
+/**
+ * Client-side page to update session info.
+ *
+ * Next-Auth does not have a simple way to do this purely from the client-side, so we
+ * use this page to check and redirect appropriately.
+ *
+ * NOTE: this does not replace doing server-side `hasPremium` checks! This is just
+ * a convenience so users do not need to sign out and back in to refresh their session
+ * after subscribing.
+ */
+export default function Subscribed() {
+  const { update } = useSession();
+  const router = useRouter();
+
+  useEffect(() => {
+    async function updateSession() {
+      try {
+        const result = await update();
+        console.debug(result, hasPremium(result?.user));
+        if (hasPremium(result?.user)) {
+          router.replace(
+            `/redesign/user/dashboard/fix/data-broker-profiles/welcome-to-premium`
+          );
+        } else {
+          router.replace(`/`);
+        }
+      } catch (ex) {
+        console.error(ex);
+        captureException(ex);
+        router.replace(`/`);
+      }
+    }
+    void updateSession();
+  }, [update, router]);
+
+  return <div>Please wait...</div>;
```

Comment:
nit: `"Please wait..."` should be localized?
And possibly use `…` (`&hellip;`) instead of `...`?

Reply:
Yes I'll put it in pending locales at the very least. This wasn't really part of the design.